### PR TITLE
[ CGAL] Update version to 5.2.2

### PR DESCRIPTION
**Pull request by maxGimeno**
**Describe the pull request**

- #### What does your PR fix?  
 Update to CGAL 5.2.2

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
 Yes

### CODE REVIEW

File: ports/cgal/CONTROL [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Source: cgal
-Version: 5.2
-Port-Version: 3
+Version: 5.2.2
+Port-Version: 4
```
Comment (ras0219) [inline ports/cgal/CONTROL:3]: Typically, when updating the version we reset the `port-version` to 0.

**Review by ras0219 — APPROVED** [review]
Thanks for the PR!

**Comment by JonLiu1993** [comment]
@maxGimeno ,Thanks for your pr.
have you tested the feature locally on x64-linux, x64-windows, and x64-windows-static？

**Comment by maxGimeno** [comment]
tested on x64-windows

**Comment by JonLiu1993** [comment]
> tested on x64-windows

Could you test x64-windows-static and x64-linux?

File: versions/c-/cgal.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "f263381044a3afdbaa4f101a0497a9317e32467d",
+      "version-string": "5.2.2",
+      "port-version": 4
```
Comment (JonLiu1993) [inline versions/c-/cgal.json:6]: When update version we should set port-version to 0

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by maxGimeno** [comment]
It doesn't work on x64-linux, at least on my machine. Mpfr doesn't build. And I don't know what `x64-windows-static` is and how to test it

File: versions/c-/cgal.json [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "8ecf63852d98760d2b2bd0c75b70ab95a5e73aca",
+      "version-string": "5.2.2",
+      "port-version": 0
+    },
+    {
+      "git-tree": "f263381044a3afdbaa4f101a0497a9317e32467d",
+      "version-string": "5.2.2",
+      "port-version": 4
+    },
```
Comment (PhoebeHui) [inline versions/c-/cgal.json:12]:
```suggestion
```
Could you please remove this?

**Review by PhoebeHui — CHANGES_REQUESTED** [review]

**Review by PhoebeHui — APPROVED** [review]
LGTM, thanks for your updates!

**Comment by ras0219-msft** [comment]
Thanks again!

**Pull request merged**